go/src/koding/klient/app/defertime.go

```diff
@@ -0,0 +1,82 @@
+package app
+
+import "time"
+
+// DeferTime allows to invoke registered function after provided duration. It
+// can cancel deferred function call even if internal timer was already started.
+type DeferTime struct {
```

Comment:
Ok, here is the use case:
All the collaborator's terminals must be disconnected when kite owner disconnects the klient. So we use `OnDisconnect` kite callback. If user goes back to klient we need to stop the timer and run it again when he disconnects.

I could use `time.AfterFunc` with extra cancellation/re-initialization logic added to `OnFirstRequest` and `OnDisconnect` kite callbacks. However, I decided to move the code to `DeferTime` structure in order to make it testable(it would be hard to write tests if I have to run the entire klient) and to remove `disconnectTimer` timer from Klient structure which is used only in callbacks. I think this simplifies Klient a bit.